File: appkit-creating-and-customizing-the-touch-bar/Objective-C/NSTouchBar Catalog/TestViewControllers/SegmentedViewController.m
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
View controller responsible for showing segmented controls in an NSTouchBar instance.
*/

#import "SegmentedViewController.h"

@implementation SegmentedViewController

// Note: This particular view controller doesn't allow customizing of its NSTouchBar instance.

- (IBAction)segmentAction:(id)sender
{
    NSSegmentedControl *segControl = (NSSegmentedControl *)sender;
    NSLog(@"segment selection = %ld", segControl.selectedSegment);
}

@end

File: appkit-creating-and-customizing-the-touch-bar/Swift/NSTouchBar Catalog/SegmentedViewController.swift
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
View controller responsible for showing segmented controls in an NSTouchBar instance.
*/

import Cocoa

class SegmentedViewController: NSViewController {

    @IBAction func segmentAction(_ sender: Any?) {
        if let segmented = sender as? NSSegmentedControl {
            print("\(#function) is called. Segment \(segmented.selectedSegment)")
        }
    }
    
}
